status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,372 @@
+Such messages MUST have a community-wide functional scope.
+A third group of messages are addressed only to the participant that generated those messages itself.
+These _self-addressed_ messages MUST have a local functional scope.
+
+If we further make a distinction between "control" and "content" messages,
+we can distinguish five distinct functional scopes.
+
+All Status messages MUST have one of these functional scopes:
+
+**Global scope**
+
+1. _Global control_: messages enabling the basic functioning of the app to control features that all app users should be able to participate in. Examples include Contact Requests, Community Invites, global Status Updates, Group Chat Invites, etc.
+2. _Global content_: messages carrying user-generated content for global functions. Examples include 1:1 chat messages, images shared over private group chats, etc.
+
+**Community scope**
+
+3. _Community control_: messages enabling the basic functioning of the app to control features _only relevant to members of a specific community_. Examples include Community Membership Updates, community Status Updates, etc.
+4. _Community content_: messages carrying user-generated content _only for members of a specific community_.
+
+**Local scope**
+
+5. _Local_: messages related to functions that are only relevant to a single user. Also known as _self-addressed messages_. Examples include messages used to exchange information between app installations, such as User Backup and Sync messages.
+
+Note that the functional scope is a logical property of Status messages.
+It SHOULD however inform the underlying [transport layer sharding](#pubsub-topics-and-sharding) and [transport layer subscriptions](#subscribing) underneath the hood.
+In general a Status client SHOULD subscribe to participate in:
+- all global functions,
+- (only) the community functions for communities of which it is a member, and
+- its own local functions.
+
+### Content topics
+
+Each Status app-level message MUST define a content topic that links messages in related app-level functions and sub-functions together.
+This MUST be based on the filter use cases for [transport layer subscriptions](#subscribing)
+and [retrieving historical messages](#retrieving-historical-messages).
+A content topic SHOULD be identical across all messages that are always part of the same filter use case (or always form part of the same content-filtered query criteria).
+In other words, the number of content topics defined in the app SHOULD match the number of filter use cases.
+For the sake of illustration, consider the following common content topic and filter use cases:
+- all messages belonging to the same 1:1 chat are always filtered together and therefore SHOULD use the same content topic (see [55/STATUS-1TO1-CHAT](../55/1to1-chat.md))
+- all messages belonging to the same Community are always filtered together and therefore SHOULD use the same content topic (see [56/STATUS-COMMUNITIES](../56/communities.md)).
```

Comment:
> all messages belonging to the same Community are always filtered together.

As part of content topic revision for communities I'd like to challenge that. But it may not be the place. I just don't want this exemplary assumption to become axiomatic.

Reply:
I see your point. I've changed both statements to a conditional ("if...Community messages are always filtered together, they SHOULD use single content topic...") in [9ce3bc5](https://github.com/vacp2p/rfc-index/pull/105/commits/9ce3bc5700324e9172729808b4af9eb68d9a9a91) to be less descriptive.